Shell/LineEditor.cpp

```diff
@@ -161,13 +165,22 @@ void LineEditor::tab_complete_other_token(String& token)
         path = g.cwd;
     }
 
+    // This is a bit naughty, but necessary without reordering the loop
+    // below. The loop terminates early (on line 189), meaning that
```

Comment:
Let's not put line numbers in comments, since they become outdated the moment someone inserts a line somewhere and forgets to update this.